### PR TITLE
feat(doctest): precision metric doctest

### PR DESCRIPTION
**Pull request by halsawadi**
Fixes part of #2265

Description:

Add a doctest for Precision in the metrics module.

Check list:

- [x] New tests are added (if a new feature is added)
- [x] New doc strings: description and/or example code are in RST format
- [x] Documentation is updated (if required)

### CODE REVIEW

**Comment by ydcjeff** [comment]
Code looks good. @halsawadi Can you change the description of the PR to something like `reference #2265` or `part of #2265` because this PR doesn't fix the issue entirely.

Thanks.

File: ignite/metrics/precision.py [hunk 1]
```diff
@@ -90,14 +90,23 @@ class Precision(_BasePrecisionRecall):
...
-        .. code-block:: python
+        .. testcode::
 
-            def thresholded_output_transform(output):
+            def thresholded_output_transform(engine, output):
                 y_pred, y = output
                 y_pred = torch.round(y_pred)
                 return y_pred, y
+            engine = Engine(thresholded_output_transform)
+            metric = Precision()
```
Comment (vfdev-5) [inline ignite/metrics/precision.py:100]: I'm not sure if this is what we want to expose to our users. `thresholded_output_transform` should remain the argument to `Precision` and not for `Engine`.

Comment (ydcjeff) [reply ignite/metrics/precision.py:100]: Actually that should be `process_function`.

File: ignite/metrics/precision.py [hunk 2]
```diff
@@ -90,14 +90,23 @@ class Precision(_BasePrecisionRecall):
         In binary and multilabel cases, the elements of `y` and `y_pred` should have 0 or 1 values. Thresholding of
         predictions can be done as below:
 
-        .. code-block:: python
+        .. testcode::
 
-            def thresholded_output_transform(output):
+            def thresholded_output_transform(engine, output):
```
Comment (ydcjeff) [inline ignite/metrics/precision.py:95]:
```suggestion
            def process_fn(engine, output):
```

File: ignite/metrics/precision.py [hunk 3]
```diff
@@ -90,14 +90,23 @@ class Precision(_BasePrecisionRecall):
...
                 y_pred, y = output
                 y_pred = torch.round(y_pred)
                 return y_pred, y
+            engine = Engine(thresholded_output_transform)
```
Comment (ydcjeff) [inline ignite/metrics/precision.py:99]:
```suggestion
            engine = Engine(process_fn)
```

File: ignite/metrics/precision.py (hunk 1 above)
Comment (vfdev-5) [reply ignite/metrics/precision.py:100]: The example shows how to binarize predictions for Precision metric only. Returning binarized preds from validator may be a limitation...

Comment (vfdev-5) [reply ignite/metrics/precision.py:100]: In general we can improve the docstring for Precision (same for Accuracy and this comment: https://github.com/pytorch/ignite/pull/2287#discussion_r735169293) and show all possible cases: 

    binary case
    multiclass case
    multilabel case

**Comment by sdesrozis** [comment]
@halsawadi Thank you for your work. Are you still working on this ?

**Comment by sdesrozis** [comment]
@halsawadi Thank you for your help. I close this PR since the doctest for Precision has been merged.

**Pull request closed without merging**